spyder/plugins/ipythonconsole.py

```diff
@@ -1050,15 +1046,194 @@ def additional_options(self):
+
+        The stylesheet can contain classes for:
+            Qt: QPlainTextEdit, QFrame, QWidget, etc
+           Pygments: .c, .k, .o, etc. (see PygmentsHighlighter)
+            IPython: .error, .in-prompt, .out-prompt, etc
+        """
+
+        def give_font_weight(is_bold):
+            if is_bold:
+                return 'bold'
+            else:
+                return 'normal'
+
+        def give_font_style(is_italic):
+            if is_italic:
+                return 'italic'
+            else:
+                return 'normal'
+
+        color_scheme = get_color_scheme(CONF.get('color_schemes', 'selected'))
+
+        fon_c, fon_fw, fon_fs = color_scheme['normal']
+        font_color =  fon_c
+        font_font_weight = give_font_weight(fon_fw)
+        font_font_style = give_font_style(fon_fs)
+        background_color = color_scheme['background']
+        selection_background_color = '#ccc'
+        error_color = 'red'
+        instance_color = color_scheme['instance'][0]
+        in_prompt_color = instance_color
+        in_prompt_number_font_weight = 'bold'
+        out_prompt_color = instance_color
+        out_prompt_number_font_weight = 'bold'
+        inverted_background_color = font_color
+        inverted_font_color = background_color
+        unmatched_parenthesis_color = color_scheme['unmatched_p']
+        matched_parenthesis_color = color_scheme['matched_p']
+
+        sheet = """QPlainTextEdit, QTextEdit {{
+                                              color: {} ;
+                                              font-weight: {};
+                                              font-style: {};
+                                              background-color: {};
+                                              selection-background-color: {}
+                                             }}
+                  .error {{ color: {}; }}
+                  .in-prompt {{ color: {}; }}
+                  .in-prompt-number {{ font-weight: {}; }}
+                  .out-prompt {{ color: {}; }}
+                  .out-prompt-number {{ font-weight: {}; }}
+                  /* .inverted is used to highlight selected completion */
+                  .inverted {{ color: {}; background-color: {}; }}
+                  """
+
+        sheet_formatted = sheet.format(font_color, font_font_weight,
+                                       font_font_style, background_color,
+                                       selection_background_color,
+                                       error_color,
+                                       in_prompt_color,
+                                       in_prompt_number_font_weight,
+                                       out_prompt_color,
+                                       out_prompt_number_font_weight,
+                                       inverted_background_color,
+                                       inverted_font_color)
+
+        return sheet_formatted
+
+    def get_syntax_style(self):

```

Comment:
All this code feels a bit messy, we will need to think of a more streamline organization when we continue to work on themes